web-console/knikubevirt/Affinity/affinity.md

```diff
@@ -0,0 +1,61 @@
+
+
+## Affinity rules - list page
+
+![Affinity list](img/D2-0-0.jpg)
+All affinity rules within the system can be viewed and managed from this page. It is located in the main navigation, under 'Workloads' 
+
+## Creating an Affinity rule from Affinity Rules page
+
+Clicking on the "Add Affinity rule" button at the top of the affinity list will open up the "Add Affinty rule" modal.
+
+![Affinity rule modal - default](img/D2-1-0.jpg)
+Going top to bottom, the user will choose to create an affinity (attachment), which is pre-selected, or change to Anti-affinity (rejection).
+Then, the user will select the resources he wishes to apply that rule to by adding their representative labels to the "Select resources by label" field.
+
+
+![Force rule](img/D2-1-1.jpg)
+Before applying, the user can check the "Force this rule" checkbox, making this rule a required condition when scheduling the resources targeted by the added labels above.
+![applying rule](img/D2-1-2.jpg)
+The user hit "Apply rule".
+![Rule added to list](img/D2-1-3.jpg)
+The user now returns to the Affinity rules list.
+
+# Affinity rules list on a resource details page
+
+![VM details page - overview tab - Affinity section](img/D1-0-0.jpg)
+Affinity rules can be viewed and managed per resource as well. The list is located in the resource details page, on the overview tab in Affinity rules section under 'Services' if those exist.
+
+## Creating an Affinity rule from a resource details page
+
+![Affinity rule modal - default](img/D1-1-0.jpg)
+
+Clicking on the "Add affinity rule" button at the top of the affinity list will open up the "Add affinty rule" modal.
+It is the same as the modal from the Affinity Rules list page except one thing: The resource itself is already added to the "Select resources by label" field.
+The resource will be mentioned by it's name. 
+Technically, this "direct selection" of this resource is not actually a direct selection. We already mentioned that selecting a resource directly is not the best way since that resource will crush and regenerate more than a few times. Instead, that selection will provide a new, unique label to the specified resource. We do not need to surface this to the user, at least not for now.
+
+![Affinity rule modal - selecting labels](img/D1-1-1.jpg)
+User enters a label. The resource name remains first in the field, even after adding labels. It will not be viewed as a label, but as a placeholder text.
+The rule is ready to be applied. The user clicks 'Apply' and returns to the resource details page, where he left off.
+
+
+## Bypassing a resource scheduling failure due to a forced Affinity rule
+
+![Resource scheduling failure - modal](img/D3-1-0.jpg)
+If a resource cannot be scheduled due to a forced affinity rule, the user will receive a notification in the same format as any other scheduling failure.
```

Comment:
I might be missing something here but will this error be shown in the list view? If so wouldn't this be a popover on the status? 
If this is meant to be seen via a toast I'm not sure we want to go from a toast notification to a modal popup for more details, feels like a bit of an odd interaction. WDYT?

Reply:
I added this screen to mention that we need to notify the user in some manner that a resource can't be scheduled, in general.
The main question here is if we want to "push" this alert at all, or just be content with displaying it somewhere on the console. This question can be answered only by understanding what type of notifications do we want to push. So, this is a very big discussion at a much larger scope.

If we are want to "push" this then the notification drawer will probably be the only method in our arsenal. (I don't know what's the status with that one)

If we want to just display it somewhere on the console, then displaying it on we will probably see it in:
Pods list view and Pod/Deployment dashboard. Really don't know what can we display on VM list view since we don't display the number of replica sets there.
We can also add a column on the Affinity list that specifies the status of the resources. That one should cover "All resources deployed", "Some resources failed to be deployed", and "Pending deployment". But I don't think it is where the user will look for to see if all of the resources are deployed, to begin with.

@matthewcarleton 
We can stop this PR and examine all these options, where does that alert appear, and do we want to "push" it or not. We can break it to a different PR.
WDYT?

Reply:
ah I see. You're right this is a larger discussion than this PR. I think we can probably note it in this PR and create an issue for it.

Reply:
[Issue](https://github.com/openshift/openshift-origin-design/issues/246) created